### ISSUE TITLE
[REQUEST] Verisure with Yale locks not reporting user who unlocked a lock?

### ISSUE THREAD

**Issue by Hedda** [persandstrom/python-verisure#177]
Using Verisure with a Yale locks (Yale Doorman) and get no information about who unlocked a lock locally via its physical keypad.

* https://www.home-assistant.io/integrations/verisure/

Like to add this as a feature request but it sounds like the Verisure ingeration no longer has any developers working on it? 

* https://community.home-assistant.io/t/verisure-2-5-4-looking-for-testers/476155

This data is available in the official Verisure app and "My Pages" website so hoping that Verisure API have this feature to allow it.

Want data for automations/notifications and to keep state of which code/person unlocked last (e.i. "Unlocked by" / “Changed by”).

Lock otherwise works fine (tough HA integration uses [Cloud Pulling](https://www.home-assistant.io/blog/2016/02/12/classifying-the-internet-of-things/#classifiers) that only pulls in new data every 60 seconds today so is slow).

Anyway, this looks to be a long missing feature as people have worked around it before:

* https://community.home-assistant.io/t/changed-by-on-verisure-lock-with-auto-lock/59272

* https://community.home-assistant.io/t/verisure-lock-status/22431/

PS: For reference; @gablin looks to be be last person who worked on new features and his fork is a few commits ahead of this:

* https://github.com/gablin/python-verisure

  * https://github.com/persandstrom/python-verisure/compare/version-2...gablin:python-verisure:version-2

**Comment by Hedda** [comment]
> PS: For reference; @gablin looks to be be last person who worked on new features and his fork is a few commits ahead of this:
> 
> * https://github.com/gablin/python-verisure
>   
>   * [version-2...gablin:python-verisure:version-2](https://github.com/persandstrom/python-verisure/compare/version-2...gablin:python-verisure:version-2)

FYI, apparently a few people are also working on a new custom Securitas component that uses the Securitas Direct API, see:

* https://github.com/guerrerotook/securitas-direct-new-api

**Comment by persandstrom** [comment]
Hi,

You get the information from the smart-lock command

```
$ vsure user@email.com password --smart-lock
```

Is your issue related that you have auto-lock enabled so that you do not see who unlocked because the door is locked again before the information is polled, or that the information is not available in home-assistant at all?

**Comment by kribjo** [comment]
I was also annoyed by this and found this feature request when researching.

I use the integration in HA and it for the *alarm* (_alarm_control_panel.verisure_alarm_) it reports something like this:
code_format: number
changed_by: John Doe
code_arm_required: true
friendly_name: Verisure Alarm
supported_features: 3

While for the *lock* (_lock.ytterdorr_) it does not report changed_by:
code_format: ^\d{4}$
method: AUTO
friendly_name: Ytterdörr
supported_features: 0

So I can get the information about who changed the alarm state, but no information who unlocked the door.
The information about who unlocked the door (by code/tag/app) is available in the Verisure app history log.

This corresponds to the documentation on https://www.home-assistant.io/integrations/verisure#alarm-control-panel and https://www.home-assistant.io/integrations/verisure#lock .

That was a long background to say that I think that the answer to the question in https://github.com/persandstrom/python-verisure/issues/177#issuecomment-2356843405 is probably that, like for me, the information is not available in HA at all.

**Comment by persandstrom** [comment]
Yes, I would assume as well that this is a HA issue, and that the feature request would be better placed on HA. The data can be retrieved by python-verisure, both the information about the last user that interacted with the door, and the event log to find previous interactions.
